spring-xsuaa/src/main/java/com/sap/cloud/security/xsuaa/XsuaaServicesParser.java

```diff
@@ -29,29 +34,87 @@
 	private static final String CREDENTIALS = "credentials";
 	private static final String VCAP_SERVICES = "VCAP_SERVICES";
 	private static final String XSUAA_TAG = "xsuaa";
+	private static final String K8S_XSUAA_PATH = "/etc/secrets/sapcp/xsuaa";
 
-	private final String vcapServices;
+	private String vcapServices;
+	private Map<String, Object> k8sServiceSecrets;
 	private JSONObject credentialsJSON;
 
 	public XsuaaServicesParser() {
-		vcapServices = System.getenv().get(VCAP_SERVICES);
-		if (vcapServices == null || vcapServices.isEmpty()) {
-			logger.warn("Cannot extract XSUAA properties from VCAP_SERVICES environment variable.");
+		if (isK8sEnv()){
+			k8sServiceSecrets = getK8sServiceSecrets();
+		} else {
+			logger.debug("CF environment detected");
+			vcapServices = System.getenv().get(VCAP_SERVICES);
+			if (vcapServices == null || vcapServices.isEmpty()) {
+				logger.warn("Cannot extract XSUAA properties from VCAP_SERVICES environment variable.");
+			}
 		}
 	}
 
 	public XsuaaServicesParser(InputStream inputStream) throws IOException {
-		vcapServices = IOUtils.toString(inputStream, Charsets.toCharset(UTF_8.name()));
-		if (vcapServices == null || vcapServices.isEmpty()) {
-			logger.warn("Cannot parse inputStream to extract XSUAA properties.");
+		if (isK8sEnv()){
```

Comment:
get rid of this `isK8sEnv()` if-else-blocks in constructor, instead introduce a new constructor that accepts the key-value pairs (Set, Map or Properties).

(Parser should not access Secret File, it should be done within XsuaaServicePropertySourceFactory.java)